File: Security_Tools/XSS ATTACK/xss.py
# هجمة xss:

import requests

url = "http://localhost/pycyber1/xss.php"
code = '<script>alert("XSS FOUND");</script>'
data = {
    "search": code, #حقن حقل الادخال search بالكود
    }
response = requests.post(url,data)

if code in response.text:
    print(f"[+] XSS Found : {url} | {code}")
else:
    print(f"[-] XSS Not Found : {url}")

File: Security_Tools/XSS ATTACK/playground/xss.php
<?php
if($_SERVER["REQUEST_METHOD"] == "POST"){
        $sercho = $_POST["search"];
        // echo "<p>".$sercho."</p>";
        // الحماية:
        $security = htmlspecialchars($sercho,ENT_QUOTES,'UTF-8'); #استغناء عن رموز javascript والمسافات..]
        echo "<p>".$security."</p>";
        // الحقنة:
        // <script>alert("XSS FOUND");</script>
}
?>
<!-- الموقع: -->

<form method="POST" >
    <input type="text" name="search" placeholder = "search about">
    <button type = "submit">Search</button>    
</form> 

